SWC-7338

jay-hodgson opened the pull request.

jay-hodgson commented on apps/SageAccountWeb/.env.development:
```diff
@@ -1,5 +1,5 @@
-VITE_SYNAPSE_REPO_ENDPOINT=https://repo-dev.dev.sagebase.org
-VITE_SYNAPSE_PORTAL_ENDPOINT=https://dev.synapse.org
+VITE_SYNAPSE_REPO_ENDPOINT=https://repo-staging.prod.sagebase.org
```
will revert before merge

nickgros replied:
You can pass a `--mode=staging` flag to `pnpm start` to tell Vite to use the `.env.staging` file.

jay-hodgson replied:
Reverted (now using flag to point to staging locally)

jay-hodgson commented on apps/SageAccountWeb/src/hooks/useMaybeForceEnable2FA.test.ts:
```diff
@@ -0,0 +1,133 @@
+
+    const hook = renderHook(() => useMaybeForceEnable2FA())
+    await waitFor(() => {
+      expect(hook.result.current.mayForceEnable2FA).toBe(true)
```
We should not redirect back to the source app if not logged in, so I think this "true" response is ok

jay-hodgson commented on apps/SageAccountWeb/src/hooks/useMaybeForceEnable2FA.test.ts:
```diff
@@ -0,0 +1,133 @@
+
+    const hook = renderHook(() => useMaybeForceEnable2FA())
+    await waitFor(() => {
+      expect(hook.result.current.mayForceEnable2FA).toBe(false)
```
Updated to have the same behavior as the TOS hooks (return true during 2fa enrollment)

jay-hodgson commented:
Updated the Account Settings page 2FA section to match the design:
<img width="767" alt="Screenshot 2025-05-29 at 9 50 30 AM" src="https://github.com/user-attachments/assets/adc38b13-09ea-4ec1-abf0-538d76b3cefd" />
Also removed functionality to activate/deactivate from this component, since activation is now required to be on this page.

nickgros commented on packages/synapse-react-client/src/components/Authentication/TwoFactorEnrollmentForm.tsx:
```diff
@@ -16,12 +26,12 @@ import TextField from '../TextField/TextField'
 /**
- * Returns a URL that can be used to generate a QR code that 2FA authenticator apps can interpret
+ * Returns a URL that can be used to generate a QR code that 2FA authenticator apps can interpret.
+ * This uses the username fro TotpSecret to create the friendly name for the account in the authenticator app.
```
```suggestion
 * This uses the username from TotpSecret to create the friendly name for the account in the authenticator app.
```
